gapis/api/cmd_id_group.go

```diff
@@ -413,18 +432,63 @@ func (c *SubCmdRoot) Insert(base []uint64, r []uint64) {
 		}
 		sg.Insert(append(base, r[0]), r[1:])
 	} else {
-		if len(c.SubGroup.Spans) == 0 {
-			c.SubGroup.Spans = Spans{&CmdIDRange{CmdID(0), id + 1}}
+		// Add subcommand one by one to the sub groups of the SubCmdRoot
+		if id > c.SubGroup.Range.End {
+			c.SubGroup.Range.End = id + 1
 		}
-		r, ok := c.SubGroup.Spans[0].(*CmdIDRange)
-		if ok {
-			r.End = id + 1
-		} else {
-			panic("This should not happen, inner-most subcommands must be added to SubCmdRoot before other nesting Sub-SubCmdRoot")
+		for i := CmdID(0); i <= id; i++ {
+			c.SubGroup.AddCommand(i)
 		}
 	}
 }
 
+func (c *SubCmdRoot) AddSubCmdMarkerGroups(base []uint64, groups []*SubCmdMarkerGroup) error {
```

Comment:
F2F discussion: 
* Documentation
* `base` parameter should be relative to c, removes the `c.Id.Contains(SubCmdIdx(base))` condition.

Reply:
Fixed in 6b24e7f